pkg/scheduler/core/generic_scheduler.go

```diff
@@ -585,10 +415,9 @@ func addNominatedPods(ctx context.Context, pr framework.PluginsRunner, nominator
 // and add the nominated pods. Removal of the victims is done by
 // SelectVictimsOnNode(). Preempt removes victims from PreFilter state and
 // NodeInfo before calling this function.
-func podPassesFiltersOnNode(
+func PodPassesFiltersOnNode(
```

Comment:
what is the plan for this function? I don't think we want to keep the dependency form the plugin to generic_scheduler.go.

Reply:
In addition to the preemption path, this function is called by the regular Filter path (`findNodesThatPassFilters()`). I think the destination should be the runtime package (pkg/scheduler/framework/runtime).

If we do so, the last dependency is `core.ErrNoNodesAvailable`, which is easier to sort out.

Reply:
hmm, we don't want plugins to have dependencies on the runtime pkg either. We should probably have an interface to invoke it to avoid that dependency.

Reply:
Let me put a TODO here for further brainstorming.